compiler/rustc_const_eval/src/interpret/intrinsics.rs

```diff
@@ -42,6 +42,48 @@ fn numeric_intrinsic<Tag>(name: Symbol, bits: u128, kind: Primitive) -> Scalar<T
     Scalar::from_uint(bits_out, size)
 }
 
+fn variance_count<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    tp_ty: Ty<'tcx>,
+) -> InterpResult<'tcx, ConstValue<'tcx>> {
+    match tp_ty.kind() {
+        // Correctly handles non-monomorphic calls, so there is no need for ensure_monomorphic_enough.
+        ty::Adt(ref adt, _) => {
+            Ok(ConstValue::from_machine_usize(adt.variants().len() as u64, &tcx))
+        }
+        ty::TyAlias(def_id, _) => {
+            let ty = tcx.type_of(def_id);
```

Comment:
Seems a bit odd to throw away information here? (Whatever the `_` is discarding.)

Reply:
I'll instead check if there is no generic otherwise I'll use `throw_inval!`.